game/menu/gamemanager.gd

```
@@ -12,9 +12,17 @@ func _ready():
 	_build_menu()
 
 
-func load_game(game_cfg: ConfigFile):
+func load_game(game_cfg: ConfigFile, display: GameDisplay):
```

This function is used to load a game. Don't misuse it to update UI.
Instead, it should return an error code.

---

Which would not work with the current processing logic. So basically create a new function _load_game_ui that checks if load_game failed and updates the display accordingly? Sounds convoluted.

---

So you'd rather require a GameDisplay anytime any game will be loaded?
That makes it more dependent on the display. What if we introduce hub worlds? 
They wouldn't contain the GameDisplay so they couldn't use this function.
How about you call load_game() from inside the GameDisplay?
Then the changes would be contained in `_build_menu()` and the GameDisplays `_on_load_pressed()`.

---

Never said i was against changing anything. The Change you are proposing is however not trivial, as it requires rewriting some of the code. Again, not against it, but sounds like more work for a really low benefit.

Might do something about it later, but not now.

---

It wasn't that much work. That is why I did a PR 20min ago with that and also those mind-bending onready vars.